src/aspire/nufft/cufinufft.py

```diff
@@ -95,7 +102,9 @@ def transform(self, signal):
         `(ntransforms, num_pts)`.
         """
 
-        if not (signal.dtype == self.dtype or signal.dtype == self.complex_dtype):
+        if (self._original_dtype == self.dtype) and not (
```

Comment:
So the purpose of this is to only raise this error if the A100 32-bit patch has *not* been activated (i.e. line 35 above) 

i.e. if we set this up as a 32 bit plan but were forced to convert to 64, and we pass in a 64 bit signal, we still want it to go ahead without complaining, even though the `_original_dtype` is 32.

Reply:
Yes, that was my intent. I'll add a comment. Fair question.

Reply:
looks good